netket/machine/torch.py

```diff
@@ -0,0 +1,122 @@
+from .abstract_machine import AbstractMachine
+import torch as _torch
+import numpy as _np
+import warnings
+
+
+def _get_number_parameters(m):
+    r"""Returns total number of variational parameters in a torch.nn.Module."""
+    return sum(
+        map(lambda p: p.numel(), filter(lambda p: p.requires_grad, m.parameters()))
+    )
+
+
+class Torch(AbstractMachine):
+    def __init__(self, module, hilbert):
+
+        self._module = _torch.jit.load(module) if isinstance(module, str) else module
+        self._module.double()
+        self._n_par = _get_number_parameters(self._module)
+
+        # TODO check that module has input shape compatible with hilbert size
+        super().__init__(hilbert)
+
+    @property
+    def parameters(self):
+        return (
+            _torch.cat(tuple(p.view(-1) for p in self._module.parameters()))
+            .detach()
+            .numpy()
+            .astype(_np.complex128)
+        )
+
+    @parameters.setter
+    def parameters(self, p):
+        if not _np.all(p.imag == 0.0):
+            warnings.warn(
+                "PyTorch machines have real parameters, imaginary part will be discarded"
+            )
+        self._torch_pars = _torch.from_numpy(p.real)
```

Comment:
`_torch_pars` probably shouldn't be an instance attribute since it's a temporary. I'd suggest using a local variable `torch_pars` instead.

Reply:
good point